controllers/iamrole_controller.go

```diff
@@ -390,6 +383,17 @@ func (r *IamroleReconciler) UpdateMeta(ctx context.Context, iamRole *iammanagerv
 	}
 }
 
+//GenerateRoleName returns a roleName that should be created in IAM
+func GenerateRoleName(iamRole iammanagerv1alpha1.Iamrole, props config.Properties) string {
+	roleName := fmt.Sprintf("%s%s%s", props.IamRolePrefix(), props.IamRoleSeparator(), iamRole.ObjectMeta.Name)
```

Comment:
There are couple of things we probably need to take care when we leave "prefix" and "separator" to an administrator.
1. Maximum length of an IAM Role name restriction. 
2. IAM Role name allowed pattern. There are only certain special characters are allowed in IAM Role
Check for RoleName in this section https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateRole.html